packages/eslint-plugin-office-addins/src/rules/load-object-before-read.ts

```diff
@@ -98,6 +103,10 @@ export = {
             return;
           }
 
+          if (loadStar) {
```

Comment:
I don't think we need the separate 'loadStar' tracking.  If I'm reading the pre-changed code right then the "*" property should be in the loadLocation map if it's there.  We should be looking for it as part of the previous if check which would also be checking the location and not just that there was a '*' used in a load somewhere.

I'm also not liking this expanding series of isolated if statements each just doing a return.  We should look at consolidating them is possible and possibly moving the 'isInsideWriteStatement' check earlier since it doesn't need the property and we can skip the extra processing.

We're getting an series of regular "if" statements here

Reply:
I agree that the chain of ifs is getting big and we shouldn't add more to it.
I added a new `hasBeenLoaded` to be reutilized and now the `loadStar` variable is not needed any longer.